packages/patterns/src/components/pattern-overrides-controls.js

```diff
@@ -71,28 +75,44 @@ function PatternOverridesControls( { attributes, setAttributes } ) {
 		return null;
 	}
 
+	const hasUnsupportedImageAttributes =
+		blockName === 'core/image' &&
+		( !! attributes.caption?.length || !! attributes.href?.length );
+
+	const helpText = hasUnsupportedImageAttributes
+		? __(
+				`Overrides currently don't support image captions or links. Remove the caption or link first before enabling overrides.`
+		  )
+		: __(
+				'Allow changes to this block throughout instances of this pattern.'
+		  );
+
 	return (
 		<>
 			<InspectorControls group="advanced">
 				<BaseControl
 					id={ controlId }
 					label={ __( 'Overrides' ) }
-					help={ __(
-						'Allow changes to this block throughout instances of this pattern.'
-					) }
+					help={ helpText }
 				>
 					<Button
 						__next40pxDefaultSize
 						className="pattern-overrides-control__allow-overrides-button"
 						variant="secondary"
 						aria-haspopup="dialog"
 						onClick={ () => {
+							if ( hasUnsupportedImageAttributes ) {
```

Comment:
Is the onClick handler called after disabling it?

Reply:
No, looks like the image block will replace the event handler with one that stops clicks. I've pushed another commit to remove this if statement